redwood/api/src/functions/getVerifiedExternalAddresses/getVerifiedExternalAddresses.ts

```diff
@@ -0,0 +1,29 @@
+import {logger} from 'src/lib/logger'
+import {db} from 'src/lib/db'
+import {CachedProfile} from 'src/services/cachedProfiles/cachedProfiles'
+
+// This endpoint is used by the Snapshot strategy named `zorro`
+export const handler = async (event, context) => {
+  const params = JSON.parse(event.body)
+  const {purposeIdentifier, externalAddresses, snapshot} = params
+
+  const connections = await db.connection.findMany({
+    where: {
+      purposeIdentifier,
+      externalAddress: {in: externalAddresses.map((str) => str.toLowerCase())},
```

Comment:
For correctness, you might want to use `canonicalizeHex` from https://github.com/zorro-project/zorro/blob/main/redwood/api/src/lib/serializers.ts#L37 for this and also when persisting "Connection"s? In practice I assume folks will give us their addresses in a consistent format so not a huge deal.

Reply:
Oh also, you should just import the top-level `isVerified` function from `services/cachedProfiles.ts`, not the `CachedProfile` object. The `CachedProfile` object expects the arguments to come in a format compatible with the GraphQL resolver spec, so it doesn't take an unwrapped profile directly.

Reply:
(that would actually create a bug because snapshot expects to receive addresses in a certain format. agree there's a more correct way to handle this, but it's also a bit more involved... think it's worth it?)

Reply:
(more generally: I'm not sure we actually want to use `canonicalizeHex` on ethereum addresses in particular, because those have a different canonical format, which almost everything we interact with will expect that we use. We should decide though what our canonical format for ethereum addresses is — it could be all lowercase, or it could be the checksummed eth address returned by e.g. https://docs.ethers.io/v5/api/utils/address/#utils-getAddress)

Reply:
Two options:
(1) use a standard canonical encoding for ethereum addresses everywhere,
(2) use our own internal unusual canonicalization, and always convert before we share externally

Reply:
Ah yeah I think option (1) is better for eth addresses specifically. Right now I think the only value we get from Cairo and want to store as a hex string is an eth address, so should probably kill the `canonicalizeHex` function and replace it with something that does the right thing for eth addresses.

Reply:
OK cool, then the only question is: what do we want for the canonical representation for Ethereum addresses?

I'd say fixed-length strings... only question is whether or not to store them with checksummed capitalization. We probably always want to *display* them with checksumming, and expose them over APIs in their checksummed form, so it might be convenient to just store them that way, too. What do you think?

Reply:
Yeah I agree, it seems more convenient to store them in the checksummed form and I don't see any downside.